Makefile

```diff
@@ -51,7 +51,31 @@ export XC_ARCH
 ARCH:=${XC_OS}_${XC_ARCH}
 export ARCH
 
-export DBUILD_ARGS=--build-arg DBUILD_DATE=${DBUILD_DATE} --build-arg DBUILD_REPO_URL=${DBUILD_REPO_URL} --build-arg DBUILD_SITE_URL=${DBUILD_SITE_URL} --build-arg ARCH=${ARCH}
+export DBUILD_ARGS=--build-arg DBUILD_DATE=${DBUILD_DATE} --build-arg DBUILD_REPO_URL=${DBUILD_REPO_URL} --build-arg DBUILD_SITE_URL=${DBUILD_SITE_URL}
+
+# Specify the name for the binaries
+UPGRADE=upgrade
+MIGRATE=migrate
+
+# If there are any external tools need to be used, they can be added by defining a EXTERNAL_TOOLS variable 
+# Bootstrap the build by downloading additional tools
+.PHONY: bootstrap
+bootstrap:
+	@for tool in  $(EXTERNAL_TOOLS) ; do \
+		echo "Installing $$tool" ; \
+		go get -u $$tool; \
+	done
+
+.PHONY: clean
+clean: 
+	@echo '--> Cleaning directory...'
+	rm -rf bin
+	rm -rf ${GOPATH}/bin/${UPGRADE}
+	rm -rf ${GOPATH}/bin/${MIGRATE}
+	rm -rf Dockerfile.upgarde
```

Comment:
Was this a typo?

```suggestion
	rm -rf Dockerfile.upgrade
```

Reply:
As the dockerfile is not dynamic removing it should not be required.